pandas/tests/groupby/test_grouping.py

```diff
@@ -414,9 +414,7 @@ def test_grouper_getting_correct_binner(self):
 
     def test_grouper_iter(self, df):
         gb = df.groupby("A")
-        msg = "DataFrameGroupBy.grouper is deprecated"
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            grouper = gb.grouper
+        grouper = gb._grouper
```

Comment:
Are these `_grouper` usages common internally? If not, I would be OK just removing these tests

Reply:
I'm thinking as long as the method exists, we should leave the tests - especially if they aren't computationally intensive. That said, it appears we never use `__iter__`, so could be removed. I'd like to handle in a follow up. I'll open an issue and list out methods we can remove.